### ISSUE TITLE
Add Cache-Control header option to Advanced Settings

### ISSUE THREAD

**Issue by bkietz** [kzahel/web-server-chrome#130]
Files are never getting cached, so they're slow to load. It would be nice to be able to set a header and allow caching for when I have massive files (these ones are taking ~5s every time).

**Comment by kzahel** [comment]
Yes caching would be good. I admit to know very little about HTTP caching. Do I send over the modified time of the file in the header, along with some kind of "check every 10min" or "check every 60day" supplemental header?

**Comment by bkietz** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#Expiration

Setting a header like `Cache-Control: public, max-age=600` should signal browsers to cache for 10min unless flushed.

Allowing the duration to be adjustable through an `<input/>` is probably a good idea too.

**Comment by jamiesanger** [comment]
Id be keen to see Last-Modified header though. Service Workers can possibly compare modified dates from an IDB or similar to only update changed (and new) files.

**Comment by kzahel** [comment]
Yeah I think if I do a sha1 or similar of the file and send an etag header
that should allow for if modified since? I'm not really clear on the
details but hopefully I should find some time to look into it
